js/views/modals/Settings/Advanced.js

```diff
@@ -0,0 +1,125 @@
+import app from '../../../app';
+import loadTemplate from '../../../utils/loadTemplate';
+import baseVw from '../../baseVw';
+import $ from 'jquery';
+
+const siblingsToggle = '[data-next-siblings-toggle]';
+const siblingsToggleInputs = `${siblingsToggle} input[type="radio"]`;
+
+function getResponseReason(resp) {
+  let reason;
+  try {
+    reason = resp.responseJSON.reason || '';
+  } catch (e) {
+    reason = '';
+  }
+  return reason;
+}
+
+export default class extends baseVw {
+  constructor(options = {}) {
+    super({
+      className: 'settingsAdvanced',
+      ...options,
+    });
+
+    this.settings = app.settings.clone();
+    this.localSettings = app.localSettings.clone();
+
+    this.listenTo(this.settings, 'sync', () => app.settings.set(this.settings.toJSON()));
+    this.listenTo(this.localSettings, 'sync',
+      () => app.localSettings.set(this.localSettings.toJSON()));
+  }
+
+  get events() {
+    return {
+      [`change ${siblingsToggleInputs}`]: 'handleNextSiblingsToggle',
+    };
+  }
+
+  handleNextSiblingsToggle(event) {
+    const t = event.target;
+    this.applyNextSiblingsToggle(t, t.value);
+  }
+
+  applyNextSiblingsToggle(t, val) {
+    const p = $(t).parents(siblingsToggle);
+    if (val === 'true') {
+      p.addClass('showNextSiblings');
+    } else {
+      p.removeClass('showNextSiblings');
+    }
+  }
+
+  getFormData(subset = this.$formFields) {
+    return super.getFormData(subset);
+  }
+
+  setAndValidate() {
+    this.localSettings.set(this.getFormData(this.$localFields), { validate: true });
+    this.settings.set(this.getFormData(), { validate: true });
+  }
+
+  startSaving() {
+    return {
+      local: this.localSettings.save(),
+      cloud: this.settings.save(),
+    };
+  }
+
+  attemptSave({ cloudFail, localFail }) {
+    if (this.localSettings.validationError || this.settings.validationError) {
+      this.trigger('saveComplete', true);
+    } else {
+      this.trigger('savingToServer');
+      const saving = this.startSaving();
+      saving.local.done(() =>
```

Comment:
Not using $.when here (as I had suggested) is probably OK in this case since the first call is saving locally. If, however, these were two server saves, I would strongly recommend $.when or $.whenAll, since that would kick off the multiple requests concurrently rather than sequentially.